NodeTypes/Field/Honeypot/Honeypot.fusion

```diff
@@ -24,12 +24,34 @@ prototype(Sitegeist.PaperTiger:Field.Honeypot) < prototype(Neos.Neos:ContentComp
                     <Neos.Fusion.Form:Input field.name={props.name + '[one]'} attributes.value="" />
                     <Neos.Fusion.Form:Input field.name={props.name + '[two]'} attributes.value={PaperTiger.timestampWithHmac()} />
                     <Neos.Fusion.Form:Input field.name={props.name + '[three]'} attributes.id={'field_' + props.name} attributes.value="" />
-                    <script>
-                    {'
-                        window.onload = function(e){
-                            document.getElementById("field_' + props.name + '").setAttribute("value","' + PaperTiger.timestampWithHmac() + '");
-                        }
-                    '}
+                    <script data-name={"field_" + props.name} data-value={PaperTiger.timestampWithHmac()}>
+                        {'
+                            const events = ["touchstart", "keydown", "mousemove", "touchmove"];
+                            const dataset = document.currentScript.dataset;
+
+                            function insertSpamProtectionValue() {
+                                const field = document.getElementById(dataset.name);
+                                const value = dataset.value;
+                                if (field && value) {
+                                    field.setAttribute("value", value);
+                                }
+                                removeEventListeners();
+                            }
+
+                            function addEventListeners() {
+                                events.forEach(event => {
+                                    document.addEventListener(event, insertSpamProtectionValue, true);
+                                });
+                            }
+
+                            function removeEventListeners() {
+                                events.forEach(event => {
+                                    document.removeEventListener(event, insertSpamProtectionValue, true);
+                                });
+                            }
+
+                            addEventListeners();
+                        '}
```

Comment:
I would use the minified version, as the function names get global variables. This could lead to unexpected behavoir.  I you want, I could create a minimal build stack with an JS file, so you can import the content of the file. Like that, we would have it a) readable and b) replaceable.

Reply:
Makes sense, will talk to our fe devs about that

Reply:
https://github.com/sitegeist/Sitegeist.PaperTiger/pull/22